src/praisonai-agents/praisonaiagents/tools/schedule_tools.py

```
@@ -17,10 +17,29 @@
+    Prefers an ``explicit`` override, then the per-turn
+    ``SessionContext.unified_user_id`` set by the bot session manager on a
+    multi-user gateway. Returns ``None`` when no identity is resolved so the
+    scheduler stores fall back to their global, single-tenant behaviour
+    (CLI / single-user deployments are unchanged).
+    """
+    if explicit:
+        return explicit
+    try:
+        from ..session.context import get_session_context
+        return get_session_context().unified_user_id or None
+    except Exception:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)